scripts/tests/rosetta-sanity.sh

```diff
@@ -0,0 +1,175 @@
+#!/bin/bash
+
+NETWORK="mainnet"
+BLOCKCHAIN="mina"
+WAIT_FOR_SYNC=false
+TIMEOUT=900
+
+declare -A mainnet
+mainnet[id]="mainnet"
+mainnet[block]="3NLaE5ygWrgssHjchYR7auQTZHveVV5au5cv5VhbWWYPdbdSm4FA"
+mainnet[address]="http://rosetta-mainnet.gcp.o1test.net"
+mainnet[account]="B62qrQKS9ghd91shs73TCmBJRW9GzvTJK443DPx2YbqcyoLc56g1ny9"
+mainnet[payment_transaction]="5JvGLZ22Pt5co9ikFhHVcewsrGNx9xwPx16oKvJ42oujZRU7Ymfh"
+mainnet[zkapp_transaction]="5Ju42hSKHMPFFuH2iar8V1scHdWET2TV8ocaazRbEea5yFWDe7RH"
+
+declare -A devnet
+devnet[id]="devnet"
+devnet[address]="http://rosetta-devnet.gcp.o1test.net"
+devnet[block]="3NLX177ZPMRfgYX6sX6tEnhb97gvjWKiivk9Fk2q8M6vHHjAQPYk"
+devnet[account]="B62qizKV19RgCtdosaEnoJRF72YjTSDyfJ5Nrdu8ygKD3q2eZcqUp7B"
+devnet[payment_transaction]="5Jumdze53X3k8rVaNQpJKdt8voGXRgVcFBZugg21FE1K7QkJBhLb"
+devnet[zkapp_transaction]="5JuJuyKtrMvxGroWyNE3sxwpuVsupvj7SA8CDX4mqWms4ZZT4Arz"
+
+DEFAULT_HEADERS=(--header "Accept: application/json" --header "Content-Type: application/json")
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --network) NETWORK="$2"; shift ;;
+        --wait-for-sync) WAIT_FOR_SYNC=true ;;
+        --timeout) TIMEOUT="$2"; shift ;;
+        --address) mainnet[address]="$2"
+                   devnet[address]="$2"
+                   shift
+                   ;;
+        -h|--help) echo "Usage: $0 [--network mainnet|devnet] [--address <address>]"; exit 0 ;;
+        *) echo "❌  Unknown parameter passed: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+function assert() {
+    local __response=$1
+    local __query=$2
+    local __success_message=$3
+    local __error_message=$4
+    
+
+    if echo $__response | jq "if ($__query) then true else false end" | grep -q true; then
+        echo "$__success_message"
+    else
+        echo "$__error_message"
+
+        echo "   Response:"
+        echo "      $( echo $__response | jq)"
+        exit 1
+    fi
+
+}
+
+
+function wait_for_sync() {
+    declare -n __test_data=$1
+
+    echo "⏳  Waiting for rosetta to sync..."
+    local start_time=$(date +%s)
+    local end_time=$((start_time + TIMEOUT))
+    local sync_status=""
+
+    while true; do
```

Comment:
Should we also introduce the max attempts here to not wait forever in case of any issue?

Reply:
Ah, missed that, thank you.